snappi/snappicommon.py

```diff
@@ -120,7 +122,101 @@ def _decode(self, dict_object):
         raise NotImplementedError()
 
 
-class SnappiObject(SnappiBase):
+class SnappiValidator(object):
+
+    _MAC_REGEX = re.compile(
+        r'[\d+a-f]{2}:' *5 + r'[\d+a-f]{2}$', re.I|re.M)
+    _IPV4_REGEX = re.compile(
+        r'[\d+a-f]{2}:' *3 + r'[\d+a-f]{2}$', re.I|re.M)
+    _IPV6_REP1 = re.compile(r'^:[\d+a-f].+', re.I|re.M)
+    _IPV6_REP2 = re.compile(r'.+[\d+a-f]:$', re.I|re.M)
+    _IPV6_REP3 = re.compile(
+        r'[\d+a-f]{1,4}:' *7 + r'[\d+a-f]{1,4}$', re.I|re.M)
```

Comment:
what about these cases?
>>> v6 = re.compile('^([0-9A-Fa-f]{0,4}[:]{1,2}){7}([0-9A-Fa-f]{0,4})$')
>>> v6.match('a::b')
>>> v6.match('::')
>>>
matches to invalid ips as well
>>> v6.match('ab:ab:ab:1234:ab12:223d:12:ab')
<re.Match object; span=(0, 29), match='ab:ab:ab:1234:ab12:223d:12:ab'>
>>> v6.match('ab:ab:ab:1234:ab12::223d:12:ab')
<re.Match object; span=(0, 30), match='ab:ab:ab:1234:ab12::223d:12:ab'>
>>> v6.match('ab:ab:ab:1234::ab12::223d:12:ab')
<re.Match object; span=(0, 31), match='ab:ab:ab:1234::ab12::223d:12:ab'>
>>>

Reply:
These can be caught as I mentioned, by ensuring that there's only one occurrence of `::`, and additionally ensuring total count of `:` is maximum 7.

Reply:
>>> v6.match(':::::::')
<re.Match object; span=(0, 7), match=':::::::'>
>>>
it will match to invalid case and also the len of ":" is 7, which becomes a bug

Reply:
okay The approach you mentioned, it is missing valid cases like '::', '::1', '1::1', '1a:123::1' and we end with regex again